packages/compartment-mapper/src/archive.js

```diff
@@ -145,13 +145,16 @@ const translateCompartmentMap = (compartments, sources, compartmentRenames) => {
       if (compartment.modules) {
         for (const name of keys(compartmentModules).sort()) {
           const module = compartmentModules[name];
-          if (module.compartment !== undefined) {
+          if (
+            module.compartment !== undefined &&
+            compartmentRenames[module.compartment] !== undefined
+          ) {
             modules[name] = {
               ...module,
               compartment: compartmentRenames[module.compartment],
             };
           } else {
-            modules[name] = module;
+            // If compartment is not defined, the module must not be part of the compartment map.
```

Comment:
Other parts of endo claim leaving a `compartments[name].modules[name].compartment` undefined is not acceptable. 
I think this is the right way to go about fixing because:
1. validation has been in place to guarantee a compartment map with this issue could not be used
2. even if there's code running afterwards that could populate the field, it has not been the case with the error I saw, so it's definitely possible for it to be overlooked (also, I didn't identify such code at first glance)